onnxruntime/test/contrib_ops/quantize_ops_test.cc

```diff
@@ -130,8 +176,41 @@ TEST(QuantizeLinearContribOpTest, QuantizeLinear_0) {
   test.Run();
 }
 
+TEST(QuantizeLinearContribOpTest, QuantizeLinear_per_tensor_float_int8) {
+  OpTester test("QuantizeLinear", 1, onnxruntime::kMSDomain);
+  std::vector<int64_t> dims{6};
+  test.AddInput<float>("x", dims, {0, 2, 3, 1000, -254, -1000});
```

Comment:
depending on how the half point rounding is handled, there should be test cases that exercise this for negative/postive float values